packages/widgets/src/weather/component.tsx

```diff
@@ -21,113 +23,123 @@ export default function WeatherWidget({ options, width }: WidgetComponentProps<"
   );
 
   return (
-    <Stack w="100%" h="100%" justify="space-around" gap={0} align="center">
-      <WeeklyForecast weather={weather} width={width} options={options} shouldHide={!options.hasForecast} />
-      <DailyWeather weather={weather} width={width} options={options} shouldHide={options.hasForecast} />
+    <Stack align="center" justify="center" gap="0" w="100%" h="100%">
+      {options.hasForecast ? (
+        <WeeklyForecast weather={weather} options={options} />
+      ) : (
+        <DailyWeather weather={weather} options={options} />
+      )}
     </Stack>
   );
 }
 
-interface DailyWeatherProps extends Pick<WidgetComponentProps<"weather">, "width" | "options"> {
-  shouldHide: boolean;
+interface WeatherProps extends Pick<WidgetComponentProps<"weather">, "options"> {
   weather: RouterOutputs["widget"]["weather"]["atLocation"];
 }
 
-const DailyWeather = ({ shouldHide, width, options, weather }: DailyWeatherProps) => {
-  if (shouldHide) {
-    return null;
-  }
-
+const DailyWeather = ({ options, weather }: WeatherProps) => {
   return (
     <>
-      <Flex
-        align="center"
-        gap={width < 120 ? "0.25rem" : "xs"}
-        justify={"center"}
-        direction={width < 200 ? "column" : "row"}
-      >
-        <WeatherIcon size={width < 300 ? 30 : 50} code={weather.current_weather.weathercode} />
-        <Title order={2}>{getPreferredUnit(weather.current_weather.temperature, options.isFormatFahrenheit)}</Title>
-      </Flex>
-
-      {width > 200 && (
-        <Group wrap="nowrap" gap="xs">
-          <IconArrowUpRight />
-          {getPreferredUnit(weather.daily.temperature_2m_max[0]!, options.isFormatFahrenheit)}
-          <IconArrowDownRight />
-          {getPreferredUnit(weather.daily.temperature_2m_min[0]!, options.isFormatFahrenheit)}
-        </Group>
-      )}
-
+      <Group className="weather-day-group" gap="1cqmin">
+        <HoverCard>
+          <HoverCard.Target>
+            <Box>
+              <WeatherIcon size="20cqmin" code={weather.current.weathercode} />
+            </Box>
+          </HoverCard.Target>
+          <HoverCard.Dropdown>
+            <WeatherDescription weatherOnly weatherCode={weather.current.weathercode} />
+          </HoverCard.Dropdown>
+        </HoverCard>
+        <Text fz="20cqmin">{getPreferredUnit(weather.current.temperature, options.isFormatFahrenheit)}</Text>
+      </Group>
+      <Space h="1cqmin" />
+      <Group className="weather-max-min-temp-group" wrap="nowrap" gap="1cqmin">
+        <IconArrowUpRight size="12.5cqmin" />
+        <Text fz="12.5cqmin">{getPreferredUnit(weather.daily[0]?.maxTemp, options.isFormatFahrenheit)}</Text>
```

Comment:
Will there always be at least one entry?

Reply:
Yes there is always an array with more than one item